File: Android/doraemonkit-no-op/src/main/java/com/didichuxing/doraemonkit/kit/core/TouchProxy.java
package com.didichuxing.doraemonkit.kit.core;

import android.view.MotionEvent;
import android.view.View;


/**
 * @author wanglikun
 * touch 事件代理 解决点击和触摸事件的冲突
 */
public class TouchProxy {

    public TouchProxy(OnTouchEventListener eventListener) {
    }

    public void setEventListener(OnTouchEventListener eventListener) {
    }


    public boolean onTouchEvent(View v, MotionEvent event) {

        return true;
    }


    public interface OnTouchEventListener {
        void onMove(int x, int y, int dx, int dy);

        void onUp(int x, int y);

        void onDown(int x, int y);
    }
}

File: Android/kotlin/doraemonkit-no-op/src/main/java/com/didichuxing/doraemonkit/kit/core/TouchProxy.kt
package com.didichuxing.doraemonkit.kit.core

/**
 * @author wanglikun
 * touch 事件代理 解决点击和触摸事件的冲突
 */
class TouchProxy(eventListener: OnTouchEventListener?) {


    interface OnTouchEventListener {
        fun onMove(x: Int, y: Int, dx: Int, dy: Int)
        fun onUp(x: Int, y: Int)
        fun onDown(x: Int, y: Int)
    }
}